concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -104,20 +104,28 @@ import Lens.Micro.Platform
 
 -- * Birk parameters
 
-data PersistentBirkParameters (av :: AccountVersion) = PersistentBirkParameters
+data PersistentBirkParameters (pv :: ProtocolVersion) = PersistentBirkParameters
     { -- |The currently-registered bakers.
-      _birkActiveBakers :: !(BufferedRef (PersistentActiveBakers av)),
+      _birkActiveBakers :: !(BufferedRef (PersistentActiveBakers (AccountVersionFor pv))),
       -- |The bakers that will be used for the next epoch.
-      _birkNextEpochBakers :: !(HashedBufferedRef (PersistentEpochBakers av)),
+      _birkNextEpochBakers :: !(HashedBufferedRef (PersistentEpochBakers (AccountVersionFor pv))),
       -- |The bakers for the current epoch.
-      _birkCurrentEpochBakers :: !(HashedBufferedRef (PersistentEpochBakers av)),
+      _birkCurrentEpochBakers :: !(HashedBufferedRef (PersistentEpochBakers (AccountVersionFor pv))),
       -- |The seed state used to derive the leadership election nonce.
-      _birkSeedState :: !SeedState
+      _birkSeedState :: !(SeedState (SeedStateVersionFor pv))
     }
     deriving (Show)
 
 makeLenses ''PersistentBirkParameters
 
+-- |Migrate a 'SeedState' between protocol versions.
+migrateSeedState :: StateMigrationParameters oldpv pv -> SeedState (SeedStateVersionFor oldpv) -> SeedState (SeedStateVersionFor pv)
+migrateSeedState StateMigrationParametersTrivial{} ss = ss
+migrateSeedState StateMigrationParametersP1P2{} ss = ss
+migrateSeedState StateMigrationParametersP2P3{} ss = ss
+migrateSeedState StateMigrationParametersP3ToP4{} ss = ss
+migrateSeedState StateMigrationParametersP4ToP5{} ss = ss
```

Comment:
Is the reason this is OK due to the fact that we reset the seed state before doing migration? If so, can we documen this here please.

Reply:
Yes.